src/rhash/hash.c

```diff
@@ -48,7 +53,34 @@ static struct rc_hash_filereader* filereader = NULL;
 
 static void* filereader_open(const char* path)
 {
+#if defined(_WIN32) && defined(_UNICODE)
```

Comment:
I don't think we want the `&& defined(_UNICODE)`. The `UNICODE` and `_UNICODE` macros only specify what to map `TCHAR` and `tcslen` to. Since you're using the `_w` functions directly, this code should not be limited to a UNICODE compile. It could also work with an MBCS compile, or an unspecified compile.

Reply:
The intent is to allow the support to not be compiled in, in case Windows versions without Unicode support (i.e. Win95/98/ME) care to be supported (granted, probably not really?), or perhaps a frontend is using this but also is not sending in UTF8 paths but rather ANSI paths (which granted, would arguably be wrong in the first place paths are specified to be utf8 in the filereader api, so maybe this isn't really a concern)

Reply:
If that's the case, you should be using `WINVER` or something similar. I still argue that this functionality could/should work even if the bulk of the product is still using the `SetWindowTextA` line of APIs.